compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -1431,7 +1432,7 @@ private void defineMembers(List<BLangTypeDefinition> typeDefNodes, SymbolEnv pkg
                     defineNode(f, objMethodsEnv);
                 });
 
-                List<String> referencedFunctions = new ArrayList<>();
+                Set<String> referencedFunctions = new HashSet<>();
```

Comment:
When you replace, ArrayList with a HashSet, you lose the insertion order. That may cause intermittent test failures. Better option is to replace it with a LinkedHashSet.

Reply:
This is only used to check duplicates and issue ```DiagnosticCode.REDECLARED_SYMBOL``` error message.
https://github.com/ballerina-platform/ballerina-lang/blob/ac9058f25bd2282a6d4a93d4905743bc026aef7c/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java#L1879

Reply:
I renamed the DS to make sure it is not mistaken with `referencedFunctions` member field in `BObjectTypeSymbol`. 
Commit : 6e55b9358886a9cc9ec6bef3fbd9a41a834bb1d2